source

```diff
@@ -88992,47 +88998,56 @@ dictionary <dfn>PromiseRejectionEventInit</dfn> : <span>EventInit</span> {
-       without, e.g., <span data-x="run the animation frame callbacks">animation frame
-       callbacks</span> interleaved). For example, a user agent might wish to coalesce timer
-       callbacks together, with no intermediate rendering updates.</p>
-      </div>
-     </li>
-
-     <li>
-      <p>If there are a <span data-x="nested browsing context">nested browsing contexts</span>
-      <var>B</var> that the user agent believes would not benefit from having their rendering
-      updated at this time, then remove from <var>docs</var> all <code>Document</code> objects whose
-      <span data-x="concept-document-bc">browsing context</span> is in <var>B</var>.</p>
-
-      <p class="note">As with <span data-x="top-level browsing context">top-level browsing
-      contexts</span>, a variety of factors can influence whether it is profitable for a browser to
-      update the rendering of <span data-x="nested browsing context">nested browsing
-      contexts</span>. For example, a user agent might wish to spend less resources rendering
-      third-party content, especially if it is not currently visible to the user or if resources are
-      constrained. In such cases, the browser could decide to update the rendering for such content
-      infrequently or never.</p>
-     </li>
+     <li id="step-7-3-rendering-opportunities">
+     <p>A <span>browsing context</span> has a <dfn>rendering opportunity</dfn> if the user agent is
+     currently able to present the contents of the <span>browsing context</span> to the user,
+     accounting for hardware refresh rate constraints and user agent throttling for performance
+     reasons, but considering content presented even if it's outside the viewport.</p>
+
+     <p class="note"><span>Browsing context</span> <span data-x="rendering opportunity">rendering
+     opportunities</span> are determined based on hardware constraints such as display refresh
+     rates and other factors such as page performance or whether the page is in the
+     background. Rendering opportunities typically occur at regular intervals. For example, if the
+     browser is attempting to achieve a 60Hz refresh rate, then rendering opportunities occur at a
+     maximum of every 60th of a second (about 16.7ms). If the browser finds that a <span>browsing
+     context</span> is not able to sustain this rate, it might drop to a more sustainable 30
+     rendering opportunities per second for that <span>browsing context</span>, rather than
+     occasionally dropping frames. (This specification does not mandate any particular model for
+     selecting rendering opportunities.)  Similarly, if a <span>browsing context</span> is not
+     visible, the user agent might decide to drop that page to a much slower 4 rendering
+     opportunities per second, or even less.</p>
+
+     <p>If there are <span data-x="browsing context">browsing contexts</span> <var>B</var> that do
```

Comment:
I think actually just putting this paragraph first would be a bit nicer. Say what you're doing, then explain it.

Reply:
Done.